pkg/meta/redis.go

```diff
@@ -2809,53 +2809,105 @@ func (m *redisMeta) ListSlices(ctx Context, slices map[Ino][]Slice, delete bool,
 		return errno(err)
 	}
 
-	return errno(m.scanDeletedSlices(ctx, func(s Slice) error {
-		slices[1] = append(slices[1], s)
+	return errno(m.scanDeletedSlices(ctx, func(ss []Slice, _ int64) (bool, error) {
+		slices[1] = append(slices[1], ss...)
 		if showProgress != nil {
-			showProgress()
+			for range ss {
+				showProgress()
+			}
 		}
-		return nil
+		return false, nil
 	}))
 }
 
-func (m *redisMeta) scanDeletedSlices(ctx context.Context, visitor func(s Slice) error) error {
-	if visitor == nil {
+func (m *redisMeta) scanDeletedSlices(ctx Context, scan deletedSliceScan) error {
```

Comment:
why not use `doCleanupDelayedSlices`

Reply:
`doCleanupDelayedSlices` is slow and we need the progress. I would like to reuse `doCleanupDelayedSlices` and change its signature to show progress.

Reply:
However, as we cannot avoid scanning all delayed slices, I think it's better to scan only once.